bolt-modules/boltlib/lib/puppet/functions/apply_prep.rb

```diff
@@ -69,13 +72,36 @@ def apply_prep(target_spec)
           end
 
           unless need_install.empty?
+            # lazy-load expensive gem code
+            require 'concurrent'
+            pool = Concurrent::ThreadPoolExecutor.new(max_threads: executor.concurrency)
```

Comment:
I don't think we need to limit the concurrency here assuming we run these through the correct executor instance.